src/Chapter3.hs

```diff
@@ -373,6 +381,22 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data Knight = Knight { knightHealth :: Int
+                     , knightAttack :: Int
+                     , knightGold :: Int
+                     } deriving (Show)
+
+data Monster = Monster { monsterHealth :: Int
+                       , monsterAttack :: Int
+                       , monsterGold :: Int
+                       } deriving (Show)
+
+fight :: Monster -> Knight -> Int
+fight Monster { monsterHealth = mHealth, monsterAttack = mAttack, monsterGold = mGold }
+      Knight { knightHealth = kHealth, knightAttack = kAttack, knightGold = kGold }
```

Comment:
Nice usage of the record pattern matching 👍🏼 

There is an extension in Haskell, that allows you to use record fields as values in the scope.

like:

```haskell
knightAndMonsterHealth Knight{..} Monster{..} = knightHealth + monsterHealth
```

If you are interested, you can learn more about this feature here: https://kodimensional.dev/recordwildcards

Thought, it might be a bit confusing for beginners, so we haven't included this in the course. This is just FYI 🙂

Reply:
oh thats cool, that could reduce amount of code 👍